automation/tests/authentication.spec.ts

```diff
@@ -1,17 +1,150 @@
-import { test, expect } from '@playwright/test';
+import { test, expect, } from '@playwright/test';
+import { AuthenticationPage } from '../pages/authentication.page.ts';
+import dotenv from 'dotenv';
+dotenv.config();
+import { loginCredentials } from '../test-data/authentication_page_data.js';
+import { EmailAuthCredential } from 'firebase/auth/web-extension';
+import { registrationData} from '../test-data/authentication_page_data.js';
+import { randomString } from '../utils/services/creatingData.ts';
+import { randomEmail } from '../utils/services/creatingData.ts';
+
+let authenticationPage: AuthenticationPage;
+
+
 
 test.beforeEach(async({page})=>{
-  await page.goto('http://localhost:5173/');
+  await page.goto('/');
+   authenticationPage = new AuthenticationPage(page);
+});
+
+test.describe('@VB-31 - Navigation to/from login page, and navigation to registration form', ()=> {
+  test('@Smoke - Login Button redirects to login page', async ({page})=> {
+    await expect(authenticationPage.pageTitle).toHaveText('Elevate Your Mood, One Track at a Time');
+    await authenticationPage.loginButton.click();
+    await expect(authenticationPage.loginPageTitle).toHaveText('Login To Your Account');
+  });
+    test('TryOutNow Button redirects to login page', async ({page})=> {
+      await authenticationPage.tryOutButton.click();
+      await expect(authenticationPage.loginPageTitle).toHaveText('Login To Your Account');
+  });
+    test('@Smoke - Exit Button redirects to guest page', async ({page})=> {
+      await authenticationPage.loginButton.click();
+      await authenticationPage.exitButton.click();
+      await expect(authenticationPage.pageTitle).toHaveText('Elevate Your Mood, One Track at a Time');
+  });
+  test('@Smoke - Verify link \Register by clicking here\ redirects to Registration Form', async ({page})=> {
+    await authenticationPage.loginButton.click();
+    await authenticationPage.linkToRegistration.click();
+    await expect(authenticationPage.registerFormTitle).toHaveText('Register New Account');
+  });
+  test('Exit Button from registration form redirects to guest page', async ({page})=> {
+    await authenticationPage.navigateToRegistrationForm();
+    await authenticationPage.exitButton.click();
+      await expect(authenticationPage.pageTitle).toHaveText('Elevate Your Mood, One Track at a Time');
+  })
 })
 
-test('Guest page loaded', async ({ page }) => {
-  const pageTitle = page.getByText('Elevate Your Mood, One Track at a Time');
-  // const pageTitle2 = page.locator('.description h1');
-  await expect(pageTitle).toHaveText('Elevate Your Mood, One Track at a Time');
+test.describe('@VB-10 - Login. Input validation popup messages', () => {
+   test('@Smoke - Successful login', async ({page}) => {
+    await authenticationPage.loginButton.click();
```

Comment:
Lines 49-50 also should be in a function in a page level as they repeat in 5 tests.